Automattic/jetpack
# Add a way for users to enter a license key

This PR adds a way for end users to be able to set a license key into Jetpack. Hosting partners and agencies have their own docs where they are instructed to directly set the correct WP option themselves. This new UI opens up that possibility to end users. This is crucial at this point in order to support hosting partners using the WHMCS (pbtFPC-Qd-p2) integration, otherwise they'll get a licenses from their hosting partner but will not be able to do anything with it.

Given the time constraints and team member availability this UI has been massively scoped down to the bare minimum. Currently as the user submits the license key Jetpack will set the correct license option. There is already code in Jetpack that listens to the update event on the license and talks to WPCOM to do what's needed to set the correct Jetpack plan based on the license. So the scope of this PR is only to show the UI and to set the option, that's it.

#### Jetpack product discussion


Project thread: pbtFPC-11l-p2

#### Does this pull request change what data or activity we track or use?


No. It only adds a way to set a wp option for the license key.

#### Testing instructions:






* Note: there is no need to have a valid Jetpack license to test this PR.
* Go to the My Plan page and check that the UI is **not** there
* Add the Code Snippets plugin and add and enable this snippet `add_filter( 'jetpack_licensing_ui_enabled', '__return_true' );`
* Go back to the My Plan page and check it looks like this:

<img width="1062" alt="Screen Shot 2021-02-22 at 18 54 05" src="https://user-images.githubusercontent.com/348248/108789206-630b3380-753f-11eb-9f73-45f3ca51c328.png">

* Enter a license (or any text) and a successful toast should appear on the top right. 
* Check the `jetpack_licenses` option (use whatever method you like) and the text you entered should be in there. If you're using the Docker setup you can run `yarn docker:wp option get jetpack_licenses` and you should see an array printed out with the license you entered earlier. If you entered more than one license then all of them should be shown there.


#### Proposed changelog entry for your changes:


* Add UI to set license key.
